tests/test_cli.py

```diff
@@ -1,19 +1,40 @@
-from click.testing import CliRunner
+from unittest.mock import ANY, mock_open
 
-from semantic_release.cli import changelog, main, print_version, publish, version
+from click.testing import CliRunner
+from mock import patch
+
+import semantic_release.cli
+from semantic_release.cli import (
+    changelog,
+    main,
+    print_version,
+    publish,
+    should_bump_version,
+    version,
+)
 from semantic_release.errors import GitError, ImproperConfigurationError
 from semantic_release.repository import ArtifactRepo
 
 from . import mock, pytest, reset_config, wrapped_config_get
 from .mocks import mock_version_file
 
 assert reset_config
+import builtins
 
 
 @pytest.fixture
 def runner():
     return CliRunner()
 
+# Only mock open for github_output_file
```

Comment:
I'm not sure I'm keen on this - I think a better strategy is to use `pytest`'s [`tmp_path`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) fixture to set up a temporary file fixture, set `GITHUB_OUTPUT` equal to the absolute path of that file, and then check the contents of the file after `publish`.

That way if we decide to refactor internally, for example using `pathlib` instead, the tests still pass - introducing special logic in this mock function doesn't feel right to me.

This might be different again if you chose to use that utility function taking a `StringIO` buffer directly in my other comment - I think both would result in tests that are slightly more robust though.